### PR TITLE
Uprate menu options (draft)

### PR DESCRIPTION
**Pull request by rd2**
First stab at measure options.

### CODE REVIEW

File: lib/measures/tbd/README.md [hunk 1]
```diff
@@ -19,44 +19,116 @@ ModelMeasure
...
+### Uprate wall construction(s)
+Uprates selected wall construction(s), to meet overall Ut target
+**Name:** uprate_walls,
 **Type:** Boolean,
 **Units:** ,
-**Required:** true,
+**Required:** false,
+**Model Dependent:** false
+
+### Uprate roof construction(s)
```
Comment (macumber) [inline lib/measures/tbd/README.md:62]: I assume the `uprate_roofs` argument doesn't do anything unless `roof_ut` is set and also that `roof_ut` doesn't do anything unless `uprate_roofs` is true.  If that is the case, you might consider removing the `uprate_roofs` argument, do not uprating if  `roof_ut` is empty, and do uprating if  `roof_ut` is not empty?

Comment (macumber) [reply lib/measures/tbd/README.md:62]: Same for floors and walls

File: lib/measures/tbd/README.md [hunk 2]
```diff
@@ -19,44 +19,116 @@ ModelMeasure
...
+### Floor Ut target (W/m2•K)
+Overall Ut target to meet for exposed floor construction(s)
+**Name:** floor_ut,
+**Type:** Double,
+**Units:** ,
+**Required:** false,
+**Model Dependent:** false
+
+### Wall construction(s) to 'uprate'
+Target 1x (or 'ALL') wall construction(s) to 'uprate'
```
Comment (macumber) [inline lib/measures/tbd/README.md:103]: Maybe mention here that `wall_ut` must be set?

File: lib/measures/tbd/measure.rb [hunk 3]
```diff
@@ -53,122 +53,296 @@ def modeler_description
...
+        lc = s.construction.get.to_LayeredConstruction.get
+        next if walls[:c].key?(lc.nameString)
+        next if roofs[:c].key?(lc.nameString)
+        next if flors[:c].key?(lc.nameString)
+        walls[:c][lc.nameString] = {a: lc.getNetArea} if type == "wall"
+        roofs[:c][lc.nameString] = {a: lc.getNetArea} if type == "roofceiling"
+        flors[:c][lc.nameString] = {a: lc.getNetArea} if type == "floor"
+      end
+
+      walls[:c] = walls[:c].sort_by{ |k,v| v[:a] }.reverse!.to_h
```
Comment (macumber) [inline lib/measures/tbd/measure.rb:123]: I think you are sorting constructions by area here?  That seems like a nice touch!

Comment (macumber) [reply lib/measures/tbd/measure.rb:123]: And the default should be the first option?

File: lib/measures/tbd/README.md (hunk 1 above)
Comment (rd2) [reply lib/measures/tbd/README.md:62]: Good idea. Easy fix.

Comment (rd2) [reply lib/measures/tbd/README.md]: Good idea to trim down the menu items. I added a (default) "NONE" string for `roof_option` (same for wall & floors).

none

I can reword the item descriptions as needed, but additions to the [Guide](https://rd2.github.io/tbd/pages/basics.html) are necessary.

File: lib/measures/tbd/README.md (hunk 2 above)
Comment (rd2) [reply lib/measures/tbd/README.md]: The added "NONE" option should do the trick.

File: lib/measures/tbd/measure.rb (hunk 3 above)
Comment (rd2) [reply lib/measures/tbd/measure.rb:130]: See Figure with "NONE" option:

1. "ALL roof constructions" (all-encompassing option - likely the most common option to be picked)
2. roof constructions in sequence, in order of prevalence (largest area first, smallest area last)
3. "NONE" as last (default) item

**Pull request merged**